control-plane/cache/consul.go

```diff
@@ -584,7 +610,7 @@ func (c *Cache) ListServices() []api.CatalogService {
 	c.cacheMutex.Lock()
 	defer c.cacheMutex.Unlock()
 
-	entries := make([]api.CatalogService, len(c.serviceCache))
+	entries := make([]api.CatalogService, 0, len(c.serviceCache))
```

Comment:
this and the above were causing issues because we later append to the slice which was leaving N null values in the slice at the beginning where N == len(c.serviceCache)